### PR TITLE
feat(spanner): New Sample for OpenTelemetry Traces

### PR DESCRIPTION
**Pull request by surbhigarg92**
Adds a sample to configure client traces with OpenTelemetry and OTLP

### CODE REVIEW

File: spanner/opentelemetry_traces/src/main/java/com/example/spanner/OpenTelemetryUsage.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        .build();
+
+    // Export to a collector that is expecting OTLP using gRPC.
+    OpenTelemetry openTelemetry = OpenTelemetrySdk.builder()
+        .setTracerProvider(sdkTracerProvider).build();
+
+    // Enable OpenTelemetry traces before Injecting OpenTelemetry
+    SpannerOptions.enableOpenTelemetryTraces();
+
+    // Inject OpenTelemetry object via Spanner options or register as GlobalOpenTelemetry.
```
Comment (dsimil) [inline spanner/opentelemetry_traces/src/main/java/com/example/spanner/OpenTelemetryUsage.java:103]: nit: Let's mention "OpenTelemetrySdk.builder()....buildAndRegisterGlobal();" in the comment to show how global registration could be done.

**Review by dsimil — CHANGES_REQUESTED** [review]

File: spanner/opentelemetry_traces/Readme.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Cloud Spanner OpenTelemetry Traces OTLP Sample
+
+## Setup
+
+This sample requires [Java](https://www.java.com/en/download/) and [Maven](http://maven.apache.org/) to run the integration test.
```
[the remark this answers is not in the document]
Comment (surbhigarg92) [reply spanner/opentelemetry_traces/Readme.md]: Done

File: spanner/opentelemetry_traces/Readme.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+1.  **Follow the set-up instructions in [the documentation](https://cloud.google.com/java/docs/setup).**
+
+2.  Enable APIs for your project.
+    [Click here](https://console.cloud.google.com/flows/enableapi?apiid=spanner.googleapis.com&showconfirmation=true)
+    to visit Cloud Platform Console and enable the Google Cloud Spanner API.
```
[the remark this answers is not in the document]
Comment (surbhigarg92) [reply spanner/opentelemetry_traces/Readme.md:12]: Added

File: spanner/opentelemetry_traces/Readme.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    to visit Cloud Platform Console and enable the Cloud Trace API.
+
+3.  Create a Cloud Spanner instance and database via the Cloud Plaform Console's
+    [Cloud Spanner section](http://console.cloud.google.com/spanner).
+
+4.  Enable application default credentials by running the command `gcloud auth application-default login`.
+
+## Run the Example
+
+1. Set up database configuration in the `OpenTelemetryUsage.java` class:
```
Comment (olavloite) [inline spanner/opentelemetry_traces/Readme.md:24]: nit: add a link to the actual file?

File: spanner/opentelemetry_traces/Readme.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    ````
+    String projectId = "my-project";
+    String instanceId = "my-instance";
+    String databaseId = "my-database";
+    ````
+
+2. Configure trace data export. You can use either the OTEL [Collector](https://opentelemetry.io/docs/collector/quick-start/ with the OTLP Exporter or the Cloud Trace Exporter. By default, the Cloud Trace Exporter is used.
```
Comment (olavloite) [inline spanner/opentelemetry_traces/Readme.md:31]: nit:
```suggestion
2. Configure trace data export. You can use either the OpenTelemetry [Collector](https://opentelemetry.io/docs/collector/quick-start/ with the OTLP Exporter or the Cloud Trace Exporter. By default, the Cloud Trace Exporter is used.
```

File: spanner/opentelemetry_traces/Readme.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+- To use OTLP Exporter, Set up the OTEL [Collector](https://opentelemetry.io/docs/collector/quick-start/) and update the OTLP endpoint in `OpenTelemetryUsage.java` class
```
Comment (olavloite) [inline spanner/opentelemetry_traces/Readme.md:33]:
```suggestion
- To use OTLP Exporter, Set up the OpenTelemetry [Collector](https://opentelemetry.io/docs/collector/quick-start/) and update the OTLP endpoint in `OpenTelemetryUsage.java` class
```

File: spanner/opentelemetry_traces/pom.xml [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+      <artifactId>opentelemetry-sdk</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.opentelemetry</groupId>
+      <artifactId>opentelemetry-sdk-trace</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>com.google.cloud.opentelemetry</groupId>
+      <artifactId>exporter-trace</artifactId>
+      <version>0.26.0</version>
```
Comment (olavloite) [inline spanner/opentelemetry_traces/pom.xml:82]: TDLR: Change to 0.30.0

This needs to be either version 0.23.0 or lower or version 0.30.0 or higher (0.30.0 is at the time of writing the highest available). The versions between those two numbers do __not__ support Java 8. This was fixed in 0.30.0.

File: spanner/opentelemetry_traces/src/main/java/com/example/spanner/OpenTelemetryUsage.java [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+
+  // TODO(developer): Replace these variables before running the sample.
+  static String projectId = "my-project";
+  static String instanceId = "my-instance";
+  static String databaseId = "my-database";
+
+  // Replace these variables to use OTLP Exporter
+  static boolean useCloudTraceExporter = true; // Replace to false for OTLP
+  static String otlpEndpoint = "http://localhost:4317"; // Replace with your OTLP endpoint
+
```
Comment (olavloite) [inline spanner/opentelemetry_traces/src/main/java/com/example/spanner/OpenTelemetryUsage.java:52]: nit: remove empty line

**Review by olavloite — APPROVED** [review]

File: spanner/opentelemetry_traces/Readme.md (hunk 4 above)
Comment (surbhigarg92) [reply spanner/opentelemetry_traces/Readme.md:24]: Link can only be added once we merge. Will do that after this PR get merged

**Review by skuruppu — APPROVED** [review]

**Review by dsimil — APPROVED** [review]

**Pull request merged**